File: tests/fixtures/sample.kt
package com.example.services

import com.example.models.User
import com.example.contracts.UserService

/**
 * User management service.
 */
class UserServiceImpl(
    private val connectionString: String
) : BaseService(), UserService {

    // Class-body property (guaranteed to be property_declaration in AST)
    private var cacheEnabled: Boolean = false

    /**
     * Creates a new user in the system.
     */
    fun createUser(name: String, email: String): User {
        return User()
    }

    fun listUsers(limit: Int): List<User> {
        return emptyList()
    }

    protected fun validateInput(input: String): Boolean {
        return true
    }

    private fun loadCache() {
    }

    companion object {
        const val MAX_RETRIES = 3
    }
}

interface UserService {
    fun createUser(name: String, email: String): User
}

data class UserConfig(
    val maxRetries: Int,
    val timeout: Long
)

sealed class Result {
    data class Success(val data: Any) : Result()
    data class Error(val message: String) : Result()
}

enum class UserStatus {
    Active,
    Inactive,
    Suspended
}

fun helperFunction(input: String): String {
    return input
}

File: tests/fixtures/sample.php
<?php

namespace App\Services;

use App\Models\User;
use App\Contracts\UserServiceInterface;

/**
 * User management service.
 */
class UserService extends BaseService implements UserServiceInterface
{
    private string $connectionString;
    public static int $maxRetries = 3;

    /**
     * Creates a new user in the system.
     */
    public function createUser(string $name, string $email): User
    {
        return new User();
    }

    public function listUsers(int $limit): array
    {
        return [];
    }

    protected function validateInput(string $input): bool
    {
        return true;
    }

    private function loadCache(): void
    {
    }
}

interface UserServiceInterface
{
    public function createUser(string $name, string $email): User;
}

trait Cacheable
{
    public function clearCache(): void
    {
    }
}

enum UserStatus: string
{
    case Active = 'active';
    case Inactive = 'inactive';
    case Suspended = 'suspended';
}

function helper_function(string $input): string
{
    return $input;
}
